### PR TITLE
Fix Point Cloud Rendering valid mask issue

### LINKED ISSUE #1201: [Bug]: valid mask of point cloud rendering is not correct
### Bug Description

When we doing point cloud rendering, some pixels might be invalid since there could be no object on that ray-casting line. The current code fails to capture this and filtering valid pixels.

### Steps to Reproduce

```python
import torch

import genesis as gs
from plyfile import PlyData, PlyElement
import numpy as np


def save_point_cloud_as_ply(points, filename):
    """
    Saves a point cloud stored in a NumPy array as a PLY file using the plyfile library.

    Args:
    - points (np.ndarray): A NumPy array of shape (N, 3) containing the point cloud, where N is the number of points.
    - filename (str): The filename of the output PLY file.
    """
    # Create a structured array for the plyfile
    dtype = [("x", "f4"), ("y", "f4"), ("z", "f4")]
    structured_array = np.array(list(map(tuple, points)), dtype=dtype)

    # Create a PlyElement from the structured array and write to file
    el = PlyElement.describe(structured_array, "vertex")
    PlyData([el]).write(filename)


def main():
    ########################## init ##########################
    gs.init(seed=0, precision="32", logging_level="debug")

    ########################## create a scene ##########################
    scene = gs.Scene(
        sim_options=gs.options.SimOptions(),
        rigid_options=gs.options.RigidOptions(enable_collision=False, gravity=(0, 0, 0)),
    )

    ########################## entities ##########################

    scene.add_entity(
        morph=gs.morphs.Mesh(
            file="meshes/sphere.obj",
            scale=0.5,
            pos=(0, 0, 0),
        ),
    )

    ########################## cameras ##########################
    cam_0 = scene.add_camera(
        res=(1600, 900),
        pos=(1.0, 1.0, 1.0),
        lookat=(0, 0, 0),
        fov=60,
        GUI=True,
    )
    scene.build()

    rgb = cam_0.render()[0]
    pc, mask = cam_0.render_pointcloud()

    save_point_cloud_as_ply(pc[mask].reshape(-1, 3), "pc_partial.ply")


if __name__ == "__main__":
    main()
```

### Expected Behavior

The saved point cloud would still have invalid points in it.

### Screenshots/Videos

_No response_

### Relevant log output

```shell

```

### Environment

- OS: [e.g. Ubuntu 24.04, Windows 11 24H2]
- GPU/CPU [e.g. A100, RTX 4090, M3pr, Intel I9-9900k, Ryzen 5900x] (N/A if no GPU/CPU)
- GPU-driver version (N/A if no GPU)
- CUDA / CUDA-toolkit version (N/A if non-Nvidia)

### Release version or Commit ID

Newest version

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by wangyian-me**
## Description
Fix the bugs of finding valid depth pixels when rendering point cloud.

## Related Issue
Fixes #1201 

## Motivation and Context
We want to get all valid points when rendering point clouds. Sometimes there are invalid pixels (when nothing in the scene in this direction). We didn't correctly filter them out previously.

## How Has This Been / Can This Be Tested?
Use pyrender.
Below is a code example:

```python
import torch

import genesis as gs
from plyfile import PlyData, PlyElement
import numpy as np


def save_point_cloud_as_ply(points, filename):
    """
    Saves a point cloud stored in a NumPy array as a PLY file using the plyfile library.

    Args:
    - points (np.ndarray): A NumPy array of shape (N, 3) containing the point cloud, where N is the number of points.
    - filename (str): The filename of the output PLY file.
    """
    # Create a structured array for the plyfile
    dtype = [("x", "f4"), ("y", "f4"), ("z", "f4")]
    structured_array = np.array(list(map(tuple, points)), dtype=dtype)

    # Create a PlyElement from the structured array and write to file
    el = PlyElement.describe(structured_array, "vertex")
    PlyData([el]).write(filename)


def main():
    ########################## init ##########################
    gs.init(seed=0, precision="32", logging_level="debug")

    ########################## create a scene ##########################
    scene = gs.Scene(
        sim_options=gs.options.SimOptions(),
        rigid_options=gs.options.RigidOptions(enable_collision=False, gravity=(0, 0, 0)),
    )

    ########################## entities ##########################

    scene.add_entity(
        morph=gs.morphs.Mesh(
            file="meshes/sphere.obj",
            scale=0.5,
            pos=(0, 0, 0),
        ),
    )

    ########################## cameras ##########################
    cam_0 = scene.add_camera(
        res=(1600, 900),
        pos=(1.0, 1.0, 1.0),
        lookat=(0, 0, 0),
        fov=60,
        GUI=True,
    )
    scene.build()

    rgb = cam_0.render()[0]
    pc, mask = cam_0.render_pointcloud()

    save_point_cloud_as_ply(pc[mask].reshape(-1, 3), "pc_partial.ply")


if __name__ == "__main__":
    main()
```

With previous versions, you will still see a large amount of points far away the ball. In the updated version, those points will disappear.

## Checklist:

- [x] I read the **CONTRIBUTING** document.
- [x] I followed the `Submitting Code Changes` section of **CONTRIBUTING** document.
- [x] I tagged the title correctly (including BUG FIX/FEATURE/MISC/BREAKING)
- [ ] I updated the [documentation](https://github.com/Genesis-Embodied-AI/genesis-doc) accordingly or no change is needed.
- [x] I tested my changes and added instructions on how to test it for reviewers.

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: genesis/vis/camera.py [hunk 1]
```diff
@@ -377,7 +379,9 @@ def backproject_depth_to_pointcloud(K: np.ndarray, depth: np.ndarray, pose, worl
             T_OPENGL_TO_OPENCV = np.array([[1, 0, 0, 0], [0, -1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]])
             cam_pose = self._rasterizer._camera_nodes[self.uid].matrix @ T_OPENGL_TO_OPENCV
 
-            pc, mask = backproject_depth_to_pointcloud(intrinsic_K, depth_arr, cam_pose, world_frame)
+            pc, mask = backproject_depth_to_pointcloud(
+                intrinsic_K, depth_arr, cam_pose, world_frame, self.near, self.far
```
Comment (YilingQiao) [inline genesis/vis/camera.py:383]: Are self.near and self.far defined?

Comment (YilingQiao) [reply genesis/vis/camera.py:383]: found
